common.lic

```diff
@@ -282,4 +282,27 @@ module DRC
       .map { |_name, properties| properties['abbrev'] }
       .each { |abbrev| fput("release #{abbrev}") }
   end
+
+  def necro_harvest_check
+    if !@necro_store
```

Comment:
There are a lot of nested if/elses here. One technique that might make this code clearer is to use `return` to remove a lot of indentation. Some like:

```
    unless @necro_store
      echo "Store material: false, dropping harvested material" if $debug_mode_ct
      bput('drop material', 'you discard it')
      return
    end

    echo "Store material: true, checking count of stored material" if $debug_mode_ct
    materials = rummage('C material', than_container)
    if materials.scan(/(?=a \S+ \S+ of \S+ material)/).count >= @necro_count
      bput('drop material', 'you discard it')
      echo "Already full on stored material, dropping harvested material." if $debug_mode_ct
      return
    end

    quality = fput('glance left')
    if quality['great'] or quality['excellent'] or quality['perfect']
      bput("put material in my #{@necro_container}", 'You put')
      echo "Storing high quality material." if $debug_mode_ct
      return
    end

    bput('drop material', 'you discard it')
    echo "dropping low quality material." if $debug_mode_ct
```

What do you think?